src/main.cpp

```diff
@@ -327,8 +328,17 @@ static std::tuple<string, string, string> _create_cert(const string &stream, con
 	// Ensure a UUID is available.
 	string uuid;
 	if (device_uuid.empty()) {
-		boost::uuids::uuid tmp = boost::uuids::random_generator()();
-		uuid = boost::uuids::to_string(tmp);
+		// Fetch from PKCS11 if available as part of the slot information
+		string slot_info = _pkcs11_tool(hsm_module, "--list-slots");
+		std::regex re("(Slot .*: )([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})");
+		std::smatch match;
+		std::regex_search(slot_info, match, re);
+		if (match.size() > 2) {
+			uuid = match.str(2);
+		} else {
+			boost::uuids::uuid tmp = boost::uuids::random_generator()();
+			uuid = boost::uuids::to_string(tmp);
+		}
```

Comment:
Looks good to me. I assume you've verified this works on a board w/o an HSM?

Reply:
Not with an empty uuid and this will indeed cause an issue since the pkcs11tool command will fail, let me fix it, thanks!